httpx/_config.py

```diff
@@ -46,151 +44,113 @@ class UnsetType:
 UNSET = UnsetType()
 
 
-def create_ssl_context(
-    cert: CertTypes | None = None,
-    verify: VerifyTypes = True,
-    trust_env: bool = True,
-    http2: bool = False,
-) -> ssl.SSLContext:
-    return SSLConfig(
-        cert=cert, verify=verify, trust_env=trust_env, http2=http2
-    ).ssl_context
-
-
-class SSLConfig:
-    """
-    SSL Configuration.
-    """
-
+class SSLContext(ssl.SSLContext):
```

Comment:
If possible I'd like this not to subclass ssl.SSLContext, so we can use pyopenssl with http3

Reply:
Ah, so...

The *type* required by the client methods should be `ssl.SSLContext`.
For example in correctly typecheck against this documented usage...

https://github.com/encode/httpx/blob/998f445c83efb140b7e9ae05c993e8f36663cba2/docs/advanced/ssl.md?plain=1#L148-L149

I think it's okay for us to have an `httpx.SSLContext()` subclass.

Reply:
@graingert Interesting...

Can we progress that separately by opening a design discussion in `httpcore` for adding `pyopenssl` support?
I'd like to understand that individual part better without risking this PR getting sidetracked.